src/commands/flutter-symbols/upload.ts

```diff
@@ -0,0 +1,468 @@
+  TYPE_DART_SYMBOLS,
+  TYPE_JVM_MAPPING,
+  VALUE_NAME_DART_MAPPING,
+  VALUE_NAME_JVM_MAPPING,
+} from './interfaces'
+
+export class UploadCommand extends Command {
+  public static usage = Command.Usage({
+    description: 'Upload symbol files for Flutter.',
+    details: `
+            This command will upload all symbol files for Flutter applications in order to symbolicate errors and
+            crash reports recieved by Datadog. This includes uploading iOS dSYMs, Proguard mapping files, and Dart
+            symbol files.
+        `,
+    examples: [
+      [
+        'Upload all symbol files from default locations',
+        'datadog-ci flutter-symbols upload --dart-symbols-location ./debug-info --service-name com.datadog.example --ios-dsyms --android-mapping',
+      ],
+    ],
+  })
+
+  private androidMapping = false
+  private androidMappingLocation?: string
+  private cliVersion: string
+  private config = {
+    apiKey: process.env.DATADOG_API_KEY,
+    datadogSite: process.env.DATADOG_SITE || 'datadoghq.com',
+  }
+  private configPath?: string
+  private dartSymbolsLocation?: string
+  private disableGit = false
+  private dryRun = false
+  private flavor = 'release'
+  private gitData?: RepositoryData
+  private iosDsyms = false
+  private iosDsymsLocation?: string
+  private maxConcurrency = 5
+  private pubspecLocation = './pubspec.yaml'
+  private repositoryUrl?: string
+  private serviceName!: string
+  private version?: string
+
+  constructor() {
+    super()
+    this.cliVersion = require('../../../package.json').version
+  }
+
+  public async execute() {
+    if (!(await this.verifyParameters())) {
+      return 1
+    }
+
+    const uploadInfo: UploadInfo[] = []
+    if (this.iosDsymsLocation) {
+      uploadInfo.push({
+        fileType: 'dSYMs',
+        location: this.iosDsymsLocation!,
+        platform: 'ios',
+      })
+    }
+    if (this.androidMappingLocation) {
+      uploadInfo.push({
+        fileType: 'Proguard Mapping File',
+        location: this.androidMappingLocation!,
+        platform: 'Android',
+      })
+    }
+    if (this.dartSymbolsLocation) {
+      uploadInfo.push({
+        fileType: 'Dart Symbol Files',
+        location: this.dartSymbolsLocation!,
+        platform: 'Flutter',
+      })
+    }
+
+    this.context.stdout.write(renderCommandInfo(this.dryRun, this.version!, this.serviceName, this.flavor, uploadInfo))
+
+    this.config = await resolveConfigFromFile(this.config, {
+      configPath: this.configPath,
+      defaultConfigPath: DEFAULT_CONFIG_PATH,
+    })
+
+    if (!this.disableGit) {
+      this.gitData = await this.getGitMetadata()
+    }
+
+    const metricsLogger = getMetricsLogger({
+      apiKey: this.config.apiKey,
+      datadogSite: this.config.datadogSite,
+      defaultTags: [
+        `version:${this.version}`,
+        `service:${this.serviceName}`,
+        `cli_version:${this.cliVersion}`,
+        'flutter:true',
+        'platform:android',
```

Comment:
I am not to fully understand the different usage cases of the command, but I was under the impression that it may also target iOS.
Is this tag always valid?

Reply:
That's a good catch.

The dSYMs sub command should create its own metrics logger for iOS, but even so, this tag probably still isn't 100% correct.

I'll actually probably replace `flutter:true` with `platform:flutter`, and add `platform:android` only during the Mapping upload.